client/routes/workflow/helpers/event-full-transforms.js

```diff
@@ -24,11 +24,19 @@ import { WORKFLOW_EVENT_TYPE } from '~constants';
 export const eventFullTransforms = {
   [WORKFLOW_EVENT_TYPE.MarkerRecorded]: d => {
     if (d.markerName === 'SideEffect') {
-      return {
-        sideEffectID: d.details[0],
-        data: JSON.tryParse(atob(d.details[1])) || d.details[1],
-        decisionTaskCompletedEventId: d.decisionTaskCompletedEventId,
-      };
+      if (Array.isArray(d.details)) {
```

Comment:
nit: whats place in the java client where it is supposed to serialize the SideEffect details?

Reply:
https://github.com/uber/cadence-java-client/blob/d7fb63c233f33449e6e80cb4ad0d50d0394360db/src/main/java/com/uber/cadence/workflow/Workflow.java#L929

Reply:
thanks, will look into it.

Reply:
You can also look at @raymondkroon PRs for cadence java sample to run this locally: